jib-core/src/main/java/com/google/cloud/tools/jib/blob/BlobDescriptor.java

```diff
@@ -84,16 +84,21 @@ public long getSize() {
    * </ol>
    */
   @Override
-  public boolean equals(Object obj) {
-    if (obj == this) {
+  public boolean equals(Object other) {
+    if (this == other) {
       return true;
     }
-    if (size < 0 || !(obj instanceof BlobDescriptor)) {
+    if (other == null) {
       return false;
     }
-
-    BlobDescriptor other = (BlobDescriptor) obj;
-    return size == other.getSize() && digest.equals(other.getDigest());
+    if (getClass() != other.getClass()) {
+      return false;
+    }
+    if (size < 0) {
```

Comment:
This looks weird. It will be right to get rid of it, but I'm unsure how this will affect the current code. Perhaps there was a reason, so until we figure that out, let's leave it there.